### PR TITLE
Remove usage of legacy queryValidated functions

### PR DESCRIPTION
**Pull request by nwalters512**
This sets us up to remove `queryValidated*` and `callValidated*` in a future PR.

### CODE REVIEW

File: apps/prairielearn/src/tests/chunks.test.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ const COURSE = {
 };
 
 function getAllChunksForCourse(course_id) {
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/chunks.test.js:45]: Should this function be async? It's returning a promise.

File: apps/prairielearn/src/tests/utils/auth.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export async function getConfiguredUser(): Promise<User> {
 }
 
 export async function getOrCreateUser(authUser: AuthUser): Promise<User> {
-  const user = await callValidatedOneRow(
+  const user = await callRow(
     'users_select_or_insert',
     [authUser.uid, authUser.name, authUser.uin, 'dev'],
     z.object({ user_id: IdSchema }),
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/utils/auth.ts:46]: This seems to be a case where the sproc returns multiple columns, but only one is used in the schema. I'm wondering if a comment is warranted here to clarify this is intended.

File: packages/migrations/src/batched-migrations/batched-migration.ts [hunk 3]
```diff
@@ -117,13 +105,13 @@ export async function updateBatchedMigrationStatus(
...
+  return await queryRow(
     sql.update_batched_migration_status,
     { id, status },
     BatchedMigrationRowSchema,
   );
 }
 
 export async function retryFailedBatchedMigrationJobs(project: string, id: string): Promise<void> {
-  await queryAsync(sql.retry_failed_jobs, { project, id });
+  await await queryAsync(sql.retry_failed_jobs, { project, id });
```
Comment (jonatanschroeder) [inline packages/migrations/src/batched-migrations/batched-migration.ts:116]: Is this a typo?

File: apps/prairielearn/src/tests/chunks.test.js (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/tests/chunks.test.js]: Changed in 1e05e18.

File: apps/prairielearn/src/tests/utils/auth.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/tests/utils/auth.ts]: I added a comment in 1e05e18.

File: packages/migrations/src/batched-migrations/batched-migration.ts (hunk 3 above)
Comment (nwalters512) [reply packages/migrations/src/batched-migrations/batched-migration.ts]: Ha, yes. Fixed in 1e05e18.

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**